testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java

```diff
@@ -0,0 +1,95 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.launcher.env.environment;
+
+import com.google.inject.Inject;
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.docker.DockerFiles.ResourceProvider;
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.env.EnvironmentProvider;
+import io.trino.tests.product.launcher.env.common.HttpProxy;
+import io.trino.tests.product.launcher.env.common.HydraIdentityProvider;
+import io.trino.tests.product.launcher.env.common.Standard;
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.common.HttpProxy.HTTP_PROXY_CONF_DIR;
+import static io.trino.tests.product.launcher.env.common.HttpProxy.PROXY;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_CONFIG_PROPERTIES;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_ETC;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvSinglenodeOauth2HttpProxyAuthentication
+        extends EnvironmentProvider
+{
+    private final PortBinder binder;
+    private final HydraIdentityProvider hydraIdentityProvider;
+    private final ResourceProvider configDir;
+
+    @Inject
+    public EnvSinglenodeOauth2HttpProxyAuthentication(
+            DockerFiles dockerFiles,
+            PortBinder binder,
+            Standard standard,
+            HydraIdentityProvider hydraIdentityProvider,
+            HttpProxy httpProxy)
+    {
+        super(standard, hydraIdentityProvider, httpProxy);
+
+        this.binder = requireNonNull(binder, "binder is null");
+        this.hydraIdentityProvider = requireNonNull(hydraIdentityProvider, "hydraIdentityProvider is null");
+        this.configDir = requireNonNull(dockerFiles, "dockerFiles is null")
+                .getDockerFilesHostDirectory("conf/environment/singlenode-oauth2-http-proxy-authn/");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(COORDINATOR, dockerContainer -> {
+            dockerContainer
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("trino/config.properties")),
+                            CONTAINER_TRINO_CONFIG_PROPERTIES)
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("trino/log.properties")),
+                            CONTAINER_TRINO_ETC + "/log.properties");
```

Comment:
Since we have moved everything to `trino` folder I guess we could copy the content of that folder here.

Reply:
I think there is no explicit support for copying whole directory

Reply:
I think we can copy them via 
```
                    .withCopyFileToContainer(
                            forHostPath(configDir.getPath("trino")),
                            CONTAINER_TRINO_ETC)
```

Reply:
done